numba/npyufunc/array_exprs.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import print_function, division, absolute_import
```

Comment:
I do think if we start having a lot of rewrites in fundamental Numba, a separate directory (package) would be appropriate.  I was also thinking, however, of array_exprs being an example of a plug-in module that contains rewrites and lowering code all in one place (similar to how numba.npyufunc.dufunc includes its own typing and lowering code).  The idea is to improve the user extension story so they can see how they can write their own modules without going into numba.typing, numba.targets, or (in the future) numba.rewrites.